In [1]:
from bs4 import BeautifulSoup as bs
import urllib
import urllib.request
import requests
import dask.dataframe as dd
import pandas as pd
import numpy as np
import csv

link_pd = pd.DataFrame(columns = ['Courses', 'links','full_links'])
get_courses_details =  pd.DataFrame(columns = ['Courses', 'full_links'])
department =  pd.DataFrame(columns = ['department_id', 'department_name'])
courses_desc =  pd.DataFrame(columns = ['courses_id', 'course_name','department_id'])

## Get Course Names

In [2]:
DOMAIN = ''
URL = 'https://catalog.colorado.edu/courses-a-z/'
FILETYPE = '.parquet'
Course_catalog = ["Accounting (ACCT)","Advertising, PR and Media Design (APRD)","Aerospace Engineering (ASEN)","Air Force Aerospace Studies - ROTC (AIRR)","Anthropology (ANTH)","Applied Math (APPM)","Arabic Languages (ARAB)","Architectural Engineering (AREN)","Architecture (ARCH)","Art Film Studies (ARTF)","Art History (ARTH)","Arts and Sciences Courses (ARSC)","Art Studio and Non-Studio (ARTS)","Asian Studies (ASIA)","Astrophysical and Planetary Sciences (ASTR)","ATLAS (ATLS)","Atmospheric and Oceanic Sciences (ATOC)","Baker Residential Academic Program (BAKR)","BCOR Applied Semester Experience (BASE)","Biochemistry (BCHM)","Biomedical Engineering (BMEN)","Business Administration (BADM)","Business Core (BCOR)","Business Environment and Policy (BPOL)","Business Law (BSLW)","Business Minor (BUSM)","Career Services (CSVC)","Center for Western Civilization (CWCV)","Center of the American West (CAMW)","Central and East European Studies (CEES)","Chemical Engineering (CHEN)","Chemistry (CHEM)","Chinese (CHIN)","Cinema Studies & Moving Image Arts (CINE)","Civil Engineering (CVEN)","Classics (CLAS)","College of Engineering and Applied Science (COEN)","College of Media, Communication & Information (CMCI)","Communication (COMM)","Communication Residential Academic Program (COMR)","Comparative Literature (COML)","Computer Science (CSCI)","Computer Science Post-Baccalaureate (CSPB)","Critical Media Practices (CMDP)","Curriculum Emphasis in Social Responsibility (CESR)","Dance (DNCE)","Danish (DANE)","Digital Humanities (DHUM)","East Asian Languages and Civilizations (EALC)","Ecology and Evolutionary Biology (EBIO)","Economics (ECON)","Education (EDUC)","Electrical and Computer Engineering (ECEN)","Energy Engineering (ENEN)","Engineering, Ethics & Society (ENES)","Engineering for Developing Communities (EDEN)","Engineering Honors (EHON)","Engineering Management (EMEN)","Engineering Plus (GEEN)","English as a Second Language (ESLG)","English (ENGL)","Entrepreneurial and Small Business Management (ESBM)","Environmental Design (ENVD)","Environmental Engineering (EVEN)","Environmental Product of Design (EPOD)","Environmental Studies (ENVS)","Environment and Sustainability (ENST)","Ethnic Studies (ETHN)","Executive MBA (MBAE)","Experience Design (TDXD)","Farrand Residential Academic Program (FARR)","Farsi (FRSI)","Finance (FNCE)","First Year Exploration (FYXP)","First Year Seminar (FYSM)","French (FREN)","Geography (GEOG)","Geological Sciences (GEOL)","German (GRMN)","Germanic and Slavic Languages and Literatures (GSLL)","Global Studies Residential Academic Program (GSAP)","Graduate School (GRAD)","Graduate Teacher Education (GRTE)","Greek Language (GREK)","Hebrew (HEBR)","Hindi/Urdu (HIND)","History (HIST)","Honors (HONR)","Humanities (HUMN)","Indonesian (INDO)","Information Management and Business Analytics (BAIM)","Information Science (INFO)","Integrative Physiology (IPHY)","Intermedia Art, Writing and Performance (IAWP)","International Affairs (IAFS)","International Business (INBU)","INVST Community Studies (INVS)","Italian (ITAL)","Japanese (JPNS)","Jewish Studies (JWST)","Journalism (JRNL)","Korean (KREN)","Landscape Architecture (LAND)","Language Technology (LGTC)","Latin American Studies (LAMS)","Latin Language (LATN)","Law School (LAWS)","Leadership Minor (LEAD)","Lesbian, Gay and Bisexual Studies (LGBT)","Libby Residential Academic Program (LIBB)","Libraries (LIBR)","Linguistics (LING)","Management (MGMT)","Marketing (MKTG)","Master of the Environment (ENVM)","Materials Science and Engineering (MSEN)","Mathematics (MATH)","MBA Advanced Electives (MBAX)","MBA Core (MBAC)","Mechanical Engineering (MCEN)","Media Research and Practice (MDRP)","Media Studies (MDST)","Medieval and Early Modern Studies (MEMS)","Military Science - Army ROTC (MILR)","Molecular, Cellular and Developmental Biology (MCDB)","MS Business Core (MSBC)","MS Business Electives (MSBX)","Museum (MUSM)","Music Electives (MUEL)","Music Ensemble (EMUS)","Music (MUSC)","Naval Science - ROTC (NAVR)","Neuroscience (NRSC)","Norlin Scholars (NRLN)","Operations and Information Management (OPIM)","Operations Management (OPMG)","Organizational Leadership (ORGL)","Organization Management (ORMG)","Outdoor Recreation (OREC)","Peace and Conflict Studies (PACS)","Performance Music (PMUS)","Philosophy (PHIL)","Physics (PHYS)","Planning and Urban Design (PLAN)","Political Science (PSCI)","Portuguese (PORT)","Presidents Leadership Class (PRLC)","Psychology (PSYC)","Real Estate (REAL)","Religious Studies (RLST)","Russian (RUSS)","Sanskrit (SNSK)","Scandinavian (SCAN)","Sewall Residential Academic Program (SEWL)","Sociology (SOCY)","Spanish (SPAN)","Speech, Language and Hearing Sciences (SLHS)","Statistics (STAT)","Sustainability and Social Innovation Residential Academic Program (SSIR)","Swedish (SWED)","Technology, Cybersecurity & Policy (CYBR)","Telecommunications (TLEN)","Theatre and Dance (THDN)","Theatre (THTR)","Thesis Music (TMUS)","Tibetan (TBTN)","Women and Gender Studies (WGST)","Writing and Rhetoric (WRTG)","Yiddish (YIDD)"]


def get_soup(url):
    return bs(requests.get(url).text, 'html.parser')

i =1
for link in get_soup(URL).find_all('a'): 
    file_link = link.get('href')
    if link.string in Course_catalog:
        link_pd.loc[i] = [link.string,file_link,'https://catalog.colorado.edu'+file_link]
        i+=1


In [3]:
department['department_name'] = link_pd["Courses"].str.split("(", n = 1, expand = True)[0]
department['department_id'] = link_pd["Courses"].str.split("(", n = 1, expand = True)[1].str.split(")", n = 1, expand = True)[0]

In [4]:
department.head()

department_id                      department_name
1          ACCT                          Accounting 
2          APRD    Advertising, PR and Media Design 
3          ASEN               Aerospace Engineering 
4          AIRR  Air Force Aerospace Studies - ROTC 
5          ANTH                        Anthropology

In [5]:
department.shape

(169, 2)

In [15]:
department_dictionary = department.to_dict('records')

In [16]:
print(department_dictionary)

[{'department_id': 'ACCT', 'department_name': 'Accounting '}, {'department_id': 'APRD', 'department_name': 'Advertising, PR and Media Design '}, {'department_id': 'ASEN', 'department_name': 'Aerospace Engineering '}, {'department_id': 'AIRR', 'department_name': 'Air Force Aerospace Studies - ROTC '}, {'department_id': 'ANTH', 'department_name': 'Anthropology '}, {'department_id': 'APPM', 'department_name': 'Applied Math '}, {'department_id': 'ARAB', 'department_name': 'Arabic Languages '}, {'department_id': 'AREN', 'department_name': 'Architectural Engineering '}, {'department_id': 'ARCH', 'department_name': 'Architecture '}, {'department_id': 'ARTF', 'department_name': 'Art Film Studies '}, {'department_id': 'ARTH', 'department_name': 'Art History '}, {'department_id': 'ARSC', 'department_name': 'Arts and Sciences Courses '}, {'department_id': 'ARTS', 'department_name': 'Art Studio and Non-Studio '}, {'department_id': 'ASIA', 'department_name': 'Asian Studies '}, {'department_id': 'AS

#### From above one can confirm there are 169 unique Departments

## Get Course Names from Course Catalog

In [17]:
i = 0
j = 0
for i in range(1, link_pd.shape[0]+1):
    url = urllib.request.urlopen(link_pd['full_links'][i])
    content = url.read()
    soup = bs(content, 'lxml')
    table = soup.findAll('div',attrs={"class":"courseblock"})

    for x in table:
        print(x.find('p').text)
        j+=1    
        get_courses_details.loc[j] = [x.find('p').text,link_pd['full_links'][i]]

ACCT 3220  (3) Corporate Financial Reporting 1
ACCT 3225  (6) Corporate Financial Reporting
ACCT 3230  (3) Corporate Financial Reporting 2
ACCT 3320  (3) Cost Management
ACCT 3440  (3) Income Taxation of Individuals
ACCT 3700  (3) Accounting in a Global Economy
ACCT 4240  (3) Advanced Financial Accounting
ACCT 4250  (3) Financial Statement Analysis
ACCT 4540  (3) Accounting Information Systems
ACCT 4620  (3) Auditing and Assurance Services
ACCT 4821  (3) Experimental Seminar: Financial Report for Complex Transactions
ACCT 4850  (3) Senior Seminar - Accounting Ethics
ACCT 4900  (1-3) Independent Study
ACCT 5100  (3) Oil and Gas Accounting
ACCT 5120  (3) Business Analytics
ACCT 5240  (3) Advanced Financial Accounting
ACCT 5250  (3) Financial Statement Analysis
ACCT 5450  (3) Income Taxation of Business Entities
ACCT 5540  (3) Accounting Information Systems
ACCT 5550  (3) Data Analytics for Accounting
ACCT 5620  (3) Auditing and Assurance Services
ACCT 5700  (3) International Accounting
A

ASEN 6028  (3) Graduate Projects II
ASEN 6037  (3) Turbulent Flows
ASEN 6050  (3) Space Instrumentation
ASEN 6055  (3) Data Assimilation & Inverse Methods for Earth & Geospace Observations
ASEN 6060  (3) Advanced Astrodynamics
ASEN 6061  (3) Molecular Gas Dynamics and DSMC
ASEN 6070  (3) Satellite Geodesy
ASEN 6080  (3) Statistical Orbit Determination
ASEN 6084  (3) Optical Multi-Target Tracking
ASEN 6090  (3) Advanced Global Navigation Satellite Systems: Software and Applications
ASEN 6091  (3) Global Navigation Satellite System (GNSS) Receiver Architecture
ASEN 6107  (3) Nonlinear Finite Element Methods
ASEN 6116  (3) Spacecraft Life Support Systems
ASEN 6216  (3) Human Operation of Aerospace Vehicles
ASEN 6265  (3) Fundamentals of Spectroscopy for Optical Remote Sensing
ASEN 6331  (3) Computational Fluid Dynamics
ASEN 6337  (3) Remote Sensing Data Analysis
ASEN 6365  (3) Lidar Remote Sensing
ASEN 6367  (3) Advanced Finite Element Methods for Plates, Shells, and Solids
ASEN 6412  (3)

APPM 1235  (4) Pre-Calculus for Engineers
APPM 1236  (1) Precalculus Work Group
APPM 1340  (4) Calculus 1 with Algebra, Part A
APPM 1345  (4) Calculus 1 with Algebra, Part B
APPM 1350  (4) Calculus 1 for Engineers
APPM 1351  (1) Calculus 1 Work Group
APPM 1360  (4) Calculus 2 for Engineers
APPM 1361  (1) Calculus 2 Work Group
APPM 1390  (1) A Game for Calculus
APPM 1650  (4) Python for Mathematical and Statistical Applications
APPM 2340  (4) Calculus 3 for Statistics and Data Science
APPM 2350  (4) Calculus 3 for Engineers
APPM 2351  (1) Calculus 3 Work Group
APPM 2360  (4) Introduction to Differential Equations with Linear Algebra
APPM 2450  (1) Calculus 3 Computer Lab
APPM 2460  (1) Differential Equations Computer Lab
APPM 2720  (1-3) Open Topics in Lower Division Applied Mathematics
APPM 2750  (4) Java: Training, Mathematical Algorithms, and Mobile Apps
APPM 3010  (3) Chaos in Dynamical Systems
APPM 3050  (3) Scientific Computing in Matlab
APPM 3170  (3) Discrete Applied Mathematics

ARCH 2100  (6) Studio 1: Foundations of Architecture
ARCH 2115  (3) Architecture Materials and Methods
ARCH 3100  (6) Studio 2: Intermediate Architecture
ARCH 3114  (3) History and Theory of Architecture 1
ARCH 3214  (3) History and Theory of Architecture 2
ARCH 4010  (5) Architectural Appreciation and Design
ARCH 4100  (6) Studio 3: Capstone in Architecture
ARCH 4115  (3) Architecture Building Technology
ARTF 5000  (3) Advanced Digital Postproduction
ARTF 5003  (3) Film and Literature
ARTF 5004  (3) Topics in Film Theory
ARTF 5010  (1-3) Topics in Film Production
ARTF 5013  (3) Film, Photography and Modernism
ARTF 5021  (3) Directing/Acting for the Camera
ARTF 5023  (3) Topics in International Cinema
ARTF 5024  (3) Advanced Research Seminar
ARTF 5040  (3) Advanced Analog Alchemy
ARTF 5043  (1-3) Topics in Cinema Critical Studies
ARTF 5105  (3) Advanced Screenwriting
ARTF 5200  (3) Flow Visualization
ARTF 5400  (3) Digital Post-Production
ARTF 5453  (3) History of Avant-Garde Film
ARTF

ARTS 5050  (3) Writing Across the Arts: Culture Writing in the 21st c
ARTS 5060  (3) Art Writing As Practice
ARTS 5075  (3) Graduate Ceramics
ARTS 5087  (3) Selected Topics in Contemporary Art
ARTS 5095  (3) Special Topics in Ceramics
ARTS 5097  (1-3) Special Topics-Non-Studio
ARTS 5104  (3) Performance/Installation
ARTS 5107  (1-3) Special Topics
ARTS 5117  (3) Graduate Art Seminar
ARTS 5118  (3) Visiting Artist Program
ARTS 5126  (3) Digital Art 2
ARTS 5130  (3) Integrated Media
ARTS 5140  (3) Integrated Arts Studio
ARTS 5150  (3) Graduate Integrated Arts
ARTS 5154  (3) Metalsmithing 1
ARTS 5161  (3) Graduate Photography
ARTS 5171  (3) New Directions in Photography
ARTS 5176  (3) New Directions in Digital Art
ARTS 5202  (3) Graduate Painting
ARTS 5246  (3) Graduate Beginning Video Production
ARTS 5303  (3-18) Graduate Relief
ARTS 5316  (3) History and Theory of Digital Art
ARTS 5346  (3) Graduate Intermediate Video Production
ARTS 5403  (3) Graduate Intaglio
ARTS 5413  (3) Graduate L

ATOC 1050  (3) Weather and the Atmosphere
ATOC 1060  (3) Our Changing Environment: El Nino, Ozone, and Climate
ATOC 1070  (1) Weather and the Atmosphere Laboratory
ATOC 2050  (3) Introduction to Atmospheric Research
ATOC 2500  (1-3) Special Topics in Atmospheric and Oceanic Sciences - Lower Division
ATOC 3050  (3) Principles of Weather
ATOC 3070  (3) Introduction to Oceanography
ATOC 3180  (3) Aviation Meteorology
ATOC 3300  (3) Analysis of Climate and Weather Observations
ATOC 3500  (3) Air Chemistry and Pollution
ATOC 3600  (3) Principles of Climate
ATOC 3720  (3) Planets and Their Atmospheres
ATOC 4020  (1) Seminar in Atmospheric and Oceanic Sciences
ATOC 4200  (3) Biogeochemical Oceanography
ATOC 4215  (3) Descriptive Physical Oceanography
ATOC 4500  (1-3) Special Topics in Atmospheric and Oceanic Sciences - Upper Division
ATOC 4550  (3) Mountain Meteorology
ATOC 4700  (3) Weather Analysis & Forecasting
ATOC 4710  (3) Introduction to Atmospheric Physics
ATOC 4720  (3) Atmospheric D

BCOR 1015  (3) The World of Business
BCOR 1025  (3) Statistical Analysis in Business
BCOR 1030  (3) Communication Strategy
BCOR 2201  (1.5) Principles of Marketing
BCOR 2202  (1.5) Principles of Organizational Behavior
BCOR 2203  (1.5) Principles of Accounting I
BCOR 2204  (1.5) Principles of Financial Management
BCOR 2205  (1.5) Introduction to Information Management and Analytics
BCOR 2206  (1.5) Principles of Operations Management
BCOR 2301  (1.5) Business Law
BCOR 2302  (1.5) Business Ethics and Social Responsibility
BCOR 2303  (1.5) Principles of Accounting II
BCOR 2304  (1.5) Strategic and Entrepreneurial Thinking
BPOL 6940  (1) Master's Candidate for Degree
BPOL 6950  (1-6) Master's Thesis
BPOL 7500  (3) Doctoral Seminar: Strategic Management 1
BPOL 7530  (3) Doctoral Seminar: Strategic Management 2
BPOL 7560  (3) Entrepreneurship, International Business and Technology Management
BPOL 8900  (1-3) Independent Study
BPOL 8990  (1-10) Doctoral Dissertation
BSLW 4120  (3) Advanced B

CHIN 1010  (5) Beginning Chinese 1
CHIN 1012  (4) Introduction to Chinese Civilization
CHIN 1020  (5) Beginning Chinese 2
CHIN 1051  (3) Masterpieces of Chinese Literature in Translation
CHIN 2110  (5) Intermediate Chinese 1
CHIN 2120  (5) Intermediate Chinese 2
CHIN 2441  (3) Film and the Dynamics of Chinese Culture
CHIN 2442  (3) Modern Chinese Media Cultures
CHIN 3110  (5) Advanced Chinese 1
CHIN 3120  (5) Advanced Chinese 2
CHIN 3200  (3) Adv Wrtg Topics on Chinese & Japanese Literature and Civilization
CHIN 3311  (3) The Dao and the World in Medieval China
CHIN 3321  (3) Political Thought in Ancient China
CHIN 3331  (3) Culture and Literature of Late Imperial China
CHIN 3333  (3) Race and Ethnicity in Chinese Literature: Sinophone Culture, Diaspora, and Identity
CHIN 3334  (3) Chinese Narrative Tradition
CHIN 3341  (3) Literature and Popular Culture in Modern China
CHIN 3342  (3) Sinophone Literature in the Contemporary World
CHIN 3351  (3) Reality and Dream in Traditional Chinese

CVEN 5537  (3) Numerical Methods in Civil Engineering
CVEN 5544  (3) Solid Waste Management and Resource Recovery
CVEN 5554  (3) Fundamentals of Air Quality Management
CVEN 5564  (3) Water Profession: Communication and Utility Finance
CVEN 5565  (3) Life-Cycle Engineering of Civil Infrastructure Systems
CVEN 5574  (3) Water Utility Management: Current Issues and Future Challenges
CVEN 5575  (3) Advanced Topics in Steel Design
CVEN 5584  (3) Water Profession: Leadership and Management
CVEN 5585  (3) Advanced Topics in Reinforced Concrete Design
CVEN 5594  (3) Water Reuse and Reclamation
CVEN 5604  (3) UV Disinfection and Advanced Oxidation
CVEN 5614  (3) Bioenergy & Bioresource Recovery
CVEN 5628  (3) Seepage and Slopes
CVEN 5678  (3) Soil Improvement and Reinforcement
CVEN 5688  (3) Environmental Geotechnics
CVEN 5708  (3) Soil Mechanics
CVEN 5718  (3) Mechanics and Dynamics of Glaciers
CVEN 5728  (3) Foundation Engineering
CVEN 5738  (3) Applied Geotechnical Analysis
CVEN 5748  (3) De

COEN 1400  (3) Project Design
COEN 1500  (1) CEAS Design Lab: Engineering Your Life
COEN 1510  (1) Self Management and Leadership Principles 1
COEN 1520  (1) Self Management and Leadership Principles 2
COEN 1550  (1) YOU'RE@CU: Undergraduate Career Seminar
COEN 1830  (1) Special Topics: Engineering First-Year Symposium
COEN 2830  (1-3) Special Topics
COEN 2850  (1-3) Independent Study
COEN 2880  (1) Fresh Start for Success
COEN 3051  (2) Leadership Seminar 1: Launching the Leadership Experience
COEN 3052  (2) Leadership Seminar 2: Leadership Experience
COEN 3053  (2) Leadership Seminar 3: ELP Synthesis and Final ePortfolio
COEN 3100  (1) Engineering Transfer Student Success Seminar
COEN 3210  (3) Climate Change and Engineering
COEN 3930  (3-6) Engineering Internship/Co-op
COEN 4000  (1-3) Global Intensive Studies
COEN 4830  (1-3) Special Topics
COEN 4850  (1-3) Independent Study
COEN 4950  (1-6) Global Engineering Internship
COEN 5000  (1-3) Global Intensive Studies
COEN 5830  (1-6) Sp

CSCI 5322  (3) Algorithmic Human-Robot Interaction
CSCI 5340  (3) Startup Essentials: Entrepreneurial Projects in Computing
CSCI 5350  (3) Entrepreneurial Projects II
CSCI 5352  (3) Network Analysis and Modeling
CSCI 5360  (3) Internet Service Provider Networks
CSCI 5380  (3) Network Virtualization and Orchestration
CSCI 5402  (3) Research Methods in Human-Robot Interaction
CSCI 5403  (3) Cybersecurity
CSCI 5413  (3) Computer Security and Ethical Hacking
CSCI 5417  (3) Information Retrieval Systems
CSCI 5423  (3) Biologically-inspired Multi-Agent Systems
CSCI 5434  (3) Probability for Computer Science
CSCI 5444  (3) Introduction to Theory of Computation
CSCI 5446  (3) Chaotic Dynamics
CSCI 5448  (3) Object-Oriented Analysis and Design
CSCI 5454  (3) Design and Analysis of Algorithms
CSCI 5502  (3) Data Mining
CSCI 5514  (3) Algorithms for Whole Genome Sequence Analysis
CSCI 5525  (3) Compiler Construction
CSCI 5535  (3) Fundamental Concepts of Programming Languages
CSCI 5548  (3) Softw

CESR 3025  (3) Essential Management Skills
CESR 3040  (3) Fundamentals of Socially Responsible Leadership
CESR 3050  (3) Sustainable Space Governance
CESR 4000  (3) Leadership Challenges
CESR 4005  (3) Business Solutions for the Developing World: Learning through Service
CESR 4010  (3) Microfinance
CESR 4130  (3) Sustainable Operations
CESR 4430  (3) Corporate Boards in Action
CESR 4440  (3) Privacy in Big Data Analytics
CESR 4821  (3) Values & Power of the Consumer in Society
CESR 4825  (3) Experimental Seminar
CESR 4826  (3) Exp. Sem-Social Entrepreneurship: Designing a Better World
CESR 4828  (3) Experimental Seminar: Corporate Boards in Action
CESR 4850  (3) The Sustainable Firm: ESG Strategies and Practice
CESR 4900  (1-3) Independent Study
DNCE 1000  (2) Beginning Contemporary Dance Technique
DNCE 1012  (2) Dance Production
DNCE 1013  (2) Dance Improvisation
DNCE 1017  (3) Dance in Popular Culture and Media
DNCE 1020  (1) Beginning Contemporary Dance with Experience
DNCE 1027  (3

EBIO 5660  (4) Insect Biology
EBIO 5700  (3) Quantitative Genetics
EBIO 5740  (3) Biology of Amphibians and Reptiles
EBIO 5750  (4) Ornithology
EBIO 5760  (4) Mammalogy
EBIO 5800  (3) Critical Thinking in Biology
EBIO 5820  (1) Graduate Writing Seminar
EBIO 5840  (1-6) Independent Study (Master's Level)
EBIO 6000  (1) Seminar: Introduction to Biological Research
EBIO 6100  (1-3) Seminar in Environmental Biology
EBIO 6120  (1-3) Seminar in Environmental Biology
EBIO 6200  (1-3) Seminar in Population Biology
EBIO 6210  (1-3) Seminar in Population Biology
EBIO 6300  (1-3) Seminar in Organismic Biology
EBIO 6840  (1-7) Independent Research (Master's Level)
EBIO 6940  (1) Master's Candidate for Degree
EBIO 6950  (1-6) Master's Thesis
EBIO 7840  (1-6) Independent Study (Doctoral Level)
EBIO 8840  (1-6) Independent Research (Doctoral Level)
EBIO 8990  (1-10) Doctoral Dissertation
ECON 1010  (1-3) Economics in Action
ECON 1078  (3) Mathematical Tools for Economists 1
ECON 1088  (3) Mathematica

EDUC 4850  (3) Teaching K-12 Mathematics: Probability & Statistics
EDUC 4901  (3) Student Teaching I
EDUC 4902  (4) Student Teaching II
EDUC 4910  (3) Peer Counseling Practicum
EDUC 4912  (1) Practicum in Teacher Education
EDUC 5001  (3) Framing Equity and Justice in the Humanities Classroom
EDUC 5002  (3) Framing Equity and Justice in the STEM Classroom
EDUC 5005  (3) Advanced Social Foundations of Education
EDUC 5010  (1) Race and Equity in Higher Education
EDUC 5011  (1) College Student Career Development
EDUC 5012  (1) Higher Education Finance
EDUC 5014  (1) Ethical Dilemmas in Higher Education
EDUC 5015  (3) International / Comparative Education
EDUC 5020  (1) College Academic Advising
EDUC 5021  (1) College Athletic Affairs
EDUC 5022  (1) LGBTQ+ Topics in Higher Education
EDUC 5023  (1) College Admissions and Enrollment
EDUC 5026  (1) Technology in Higher Education
EDUC 5027  (1) Institutional Advancement and Development in Higher Education
EDUC 5028  (1) International Education 

ECEN 1030  (1-4) Special Topics
ECEN 1100  (1) Exploring ECE
ECEN 1310  (4) C Programming for ECE
ECEN 1400  (3) Introduction to Digital and Analog Electronics
ECEN 1500  (3) Sustainable Energy
ECEN 1840  (1-6) Independent Study
ECEN 2010  (1-5) Special Topics
ECEN 2020  (1-5) Special Topics
ECEN 2050  (1-5) Special Topics
ECEN 2060  (3) Special Topics
ECEN 2250  (3) Introduction to Circuits and Electronics
ECEN 2260  (3) Circuits as Systems
ECEN 2270  (3) Electronics Design Lab
ECEN 2350  (4) Digital Logic
ECEN 2360  (3) Programming Digital Systems
ECEN 2370  (3) Embedded Software Engineering
ECEN 2410  (3) Renewable Sources and Efficient Electrical Energy Systems
ECEN 2420  (3) Electronics for Wireless Systems
ECEN 2440  (3) Application of Embedded Systems
ECEN 2450  (3) Electronic and Semiconductor Device Laboratory
ECEN 2703  (3) Discrete Mathematics for Computer Engineers
ECEN 2830  (1-5) Special Topics
ECEN 2840  (1-6) Independent Study
ECEN 3002  (3) Special Topics
ECEN 3003  (3

ECEN 5793  (3) Secure Computer Architecture
ECEN 5797  (3) Introduction to Power Electronics
ECEN 5803  (3) Mastering Embedded Systems Architecture
ECEN 5807  (3) Modeling and Control of Power Electronic  Systems
ECEN 5813  (3) Principles of Embedded Software
ECEN 5817  (3) Resonant and Soft-Switching Techniques in Power Electronics
ECEN 5821  (3) Neural Systems and Physiological Control
ECEN 5823  (3) Internet of Things Embedded Firmware
ECEN 5827  (3) Analog IC Design
ECEN 5828  (3) Hybrid Dynamical Systems: Theory and Applications
ECEN 5830  (3) Special Topics
ECEN 5833  (3) Low Power Embedded Design Techniques
ECEN 5837  (3) Mixed-Signal IC Design Lab
ECEN 5840  (1-6) Independent Study
ECEN 5853  (3) Embedding Sensors and Motors
ECEN 5857  (3) Digital Control for Power Electronics
ECEN 5863  (3) Programmable Logic Embedded System Design
ECEN 5907  (3) Special Topics
ECEN 5915  (3) Foundations of Quantum Engineering
ECEN 5930  (1-3) Industry Internship
ECEN 5933  (3) Engineering Gen

ENGL 4513  (3) British Medieval Literature
ENGL 4514  (3) Advanced Topics: The Long 18th Century
ENGL 4524  (3) Advanced Topics: Romanticism
ENGL 4624  (3) Topics in Transnational Literature 1660-1900
ENGL 4634  (3) Advanced Topics: The Victorians
ENGL 4655  (3) Special Topics in American Literature to 1900
ENGL 4665  (3) Special Topics in American Literature after 1900
ENGL 4677  (3) Jewish-American Literature
ENGL 4693  (3) Advanced Topics in British Literature to 1660
ENGL 4697  (3) Special Topics in Ethnic US Literatures
ENGL 4717  (3) Native American and Indigenous Studies Capstone Seminar
ENGL 4820  (3) Honors Seminar
ENGL 4830  (3) Honors Thesis
ENGL 4840  (1-3) Independent Study---Upper Division
ENGL 4850  (1-3) Independent Study---Upper Division
ENGL 5000  (3) Introduction to Applied Shakespeare
ENGL 5003  (3) Old English 1: Introduction to Old English
ENGL 5013  (3) Old English 2: Intermediate Old English
ENGL 5019  (3) Survey of Contemporary Literary and Cultural Theory
ENGL

EPOD 2004  (3) History and Theory of Environmental Products of Design
EPOD 2100  (6) Studio 1: Foundations of Environmental Products of Design
EPOD 3100  (6) Studio 2: Intermediate Environmental Products of Design
EPOD 3101  (3) Theory and Ethics in Design
EPOD 3105  (3) Human Centered Design and Entrepreneurship Strategies
EPOD 4012  (3) EPOD Capstone Proposal Seminar
EPOD 4100  (6) Studio 3: Capstone in Environmental Products of Design
EPOD 4115  (3) Materials and Manufacturing Processes in Product Design
ENVS 1000  (4) Introduction to Environmental Studies
ENVS 1001  (4) Introduction to Human Dimensions of Environmental Studies
ENVS 1150  (3) First-Year Writing in Energy, Environment and Sustainability
ENVS 2000  (4) Applied Ecology for Environmental Studies
ENVS 2001  (3) Topical Seminar in Enivronmental Studies
ENVS 2030  (3) Introduction to Human Dimensions of the Environment
ENVS 2100  (2-4) Topics in Applied Environmental Studies
ENVS 2840  (1-6) Independent Study
ENVS 3001  (3

MBAE 6001  (1.5) Ethics I
MBAE 6010  (1.5) The Economics of Pricing, Price Forecasting, and the Macroeconomy
MBAE 6021  (1.5) Financial Accounting I
MBAE 6022  (1.5) Financial Accounting II
MBAE 6050  (3) Executive Level Strategy
MBAE 6051  (3) Operations Management
MBAE 6065  (3) Business Finance
MBAE 6081  (1.5) Data and Uncertainty
MBAE 6091  (1.5) Strategic Marketing I
MBAE 6092  (1.5) Strategic Marketing II
MBAE 6100  (3) Entrepreneurship
MBAE 6211  (1.5) Applied Financial Management I
MBAE 6470  (3) Creating Value Through Innovation
MBAE 6560  (1.5) Executive Leadership
MBAE 6565  (1.5) Inclusive Leadership
MBAE 6581  (1.5) Aligning People with Business Strategy I
MBAE 6582  (1.5) Aligning People with Business Strategy II
TDXD 5005  (3) Design Theory
TDXD 5105  (3) Collaboratory in Experience Design 1
TDXD 5500  (3) Experience Design Atelier 1: Design Evolution and Expression
TDXD 5700  (3) Experience Design Atelier 2: Introduction to Design Graphics
TDXD 5805  (3) Professional P

GEOG 1001  (4) Environmental Systems: Climate and Vegetation
GEOG 1011  (4) Environmental Systems: Landscapes and Water
GEOG 1962  (3) Geographies of Global Change
GEOG 1972  (3) Environment-Society Geography
GEOG 1982  (3) Global Geographies: Societies, Places, Connections
GEOG 1992  (3) Human Geographies
GEOG 2053  (3) Mapping a Changing World
GEOG 2092  (3) Advanced Introduction to Human Geography
GEOG 2271  (3) Introduction to the Arctic Environment
GEOG 2692  (3) Foundations in Public Health
GEOG 2852  (3) Contemporary Southeast Asia: Environmental Politics
GEOG 3022  (3) Climate Politics and Policy
GEOG 3023  (4) Statistics and Geographic Data
GEOG 3053  (4) Geographic Information Science: Mapping
GEOG 3251  (3) Mountain Geography
GEOG 3301  (3) Analysis of Climate and Weather Observations
GEOG 3351  (3) Biogeography
GEOG 3402  (3) Natural Hazards
GEOG 3412  (3) Conservation Practice and Resource Management
GEOG 3422  (3) Political Ecology
GEOG 3511  (4) Introduction to Hydrology

GEOL 5555  (3) Topics in Macroevolution
GEOL 5611  (3) Organic Geochemistry
GEOL 5612  (3) Techniques in Organic Geochemistry
GEOL 5660  (3) Sedimentology & Geobiology of Carbonates
GEOL 5670  (3) Isotope Geology
GEOL 5675  (3) Stable Isotopes in Paleoclimate and Paleoecology
GEOL 5690  (3) Tectonic History of the Western United States
GEOL 5700  (1-4) Geological Topics Seminar
GEOL 5701  (2) Super-Problems in Quatrnary Climate
GEOL 5702  (1-3) Geomorphology Seminar
GEOL 5703  (1) Seminar in Tectonics
GEOL 5704  (1) Carbonates Seminar
GEOL 5705  (1-3) Seminar in Paleoclimate
GEOL 5711  (1-3) Igneous and Metamorphic Field Geology
GEOL 5712  (1-3) Structural Field Geology
GEOL 5714  (2) Field Geophysics
GEOL 5715  (1-3) Field Techniques in Surficial Geology and Geohydrology
GEOL 5716  (1-3) Environmental Field Geochemistry
GEOL 5717  (2) Field Seminar in Geology and Tectonics
GEOL 5719  (2) Field Analysis and Tectonics of Crystalline Rocks
GEOL 5725  (1-4) Field Based Special Topics in G

HIND 1010  (5) Beginning Hindi 1
HIND 1011  (3) Introduction to South Asian Civilizations
HIND 1020  (5) Beginning Hindi 2
HIND 2110  (5) Intermediate Hindi 1
HIND 2120  (5) Intermediate Hindi 2
HIND 3110  (3) Advanced Hindi 1
HIND 3120  (3) Advanced Hindi 2
HIND 3400  (3) Special Topics
HIND 3441  (3) Screening India: A History of Bollywood Cinema
HIND 3651  (3) Living Indian Epics: The Ramayana and the Mahabharata in the Modern Political Imagination
HIND 3661  (3) South Asian Diasporas: Imagining Home Abroad
HIND 3811  (3) The Power of the Word: Subversive and Censored 20th Century Indo-Pakistani Literature
HIND 3831  (3) The Many Faces of Krishna in South Asia Literature and Culture
HIND 3851  (3) Devotional Literature in South Asia
HIND 4900  (1-3) Independent Study
HIST 1011  (3) Greeks, Romans, Kings & Crusaders: European History to 1600
HIST 1012  (3) Empire, Revolution and Global War: European History Since 1600
HIST 1015  (3) American History to 1865
HIST 1018  (3) Introductio

HIST 4738  (3) History of Early Modern Japan (1590-1868)
HIST 4758  (3) The History of Postwar Japan, 1945 to Present
HIST 4761  (3) Roman Law
HIST 4776  (3) History and Genealogy in American Society
HIST 4800  (3) Special Topics in Global History
HIST 4803  (3) Special Topics in European History
HIST 4806  (3) Special Topics in American History
HIST 4808  (3) Special Topics in World Areas History
HIST 4820  (3) Human Rights: Historical Perspectives
HIST 4827  (3) Modern U.S. Jewish History since 1880
HIST 4830  (3) Human Trafficking in Global Perspective
HIST 4837  (3) Jews in the American West
HIST 4930  (1-6) History Internship
HIST 4990  (1-3) History Lab: Methods, Sources, and Practices of the Past
HIST 5000  (3) Historical Methods: Introduction to the Professional Study of History
HIST 5012  (3) Graduate Colloquium in European History
HIST 5013  (3) Law and Society in Premodern England to 1688
HIST 5053  (3) Britain and the Empire, 1688-1964
HIST 5061  (3) Twilight of Antiquity
H

INDO 1110  (3) Beginning Indonesian 1- DILS
INDO 1120  (3) Beginning Indonesian 2
INDO 2110  (3) Intermediate Indonesian 1- DILS
INDO 2120  (3) Intermediate Indonesian 2- DILS
BAIM 2999SA  (0.1-99) BAIM Lower Division
BAIM 3100  (3) Data Visualization
BAIM 3205  (3) Business Data Management
BAIM 3210  (3) Low Code for Citizen Developers
BAIM 3220  (3) Introduction to Python Programming
BAIM 3230  (3) Enterprise Integration Technologies
BAIM 4065  (3) Leadership in a Digital Age
BAIM 4090  (3) IT and Business Strategy
BAIM 4120  (3) Business Analytics
BAIM 4200  (3) Advanced Business Analytics
BAIM 4230  (3) Customer Success with CRM
BAIM 4240  (3) Technology Integration
BAIM 4250  (3) Information Security Management
BAIM 4900  (1-3) Independent Study
INFO 1101  (3) Computation in Society
INFO 1111  (4) Introduction to Information Science: Understanding the World Through Data
INFO 1121  (4) Designing Interactions
INFO 1201  (4) Computational Reasoning 1: Information Transformation
INFO 

IAFS 1000  (4) Global Issues and International Affairs
IAFS 3000  (3) Special Topics in International Affairs
IAFS 3010  (3) Islam, Geopolitics and Society: Gender, Identity and Place
IAFS 3500  (3) French Connections: Contemporary France and America in Historical Context
IAFS 3520  (6) Global Seminar: Justice, Human Rights and Democracy in Israel
IAFS 3530  (3) Global Seminar: Jews and Muslims - The Multiethnic History of Istanbul
IAFS 3540  (3) Migration, Human Rights, and Conflict in the Mediterranean
IAFS 3600  (3) Contemporary Jewish Societies
IAFS 3610  (3) Topics in International Affairs and Jewish Studies
IAFS 3621  (3) Rogues to Revolutionaries: Russian Rebels, Past and Present
IAFS 3630  (3) Radical Nationalism in Contemporary Northern Europe
IAFS 3631  (3) Arctic Society and Culture
IAFS 3632  (3) Scandinavia and the European Union
IAFS 3640  (3) Data Analysis for Global Environmental Affairs
IAFS 3650  (3) History of Arab-Israeli Conflict
IAFS 3670  (3) Cities of the Global

JWST 1040  (3) Beginning Biblical Hebrew, Second Semester
JWST 1234  (3) Mysticism and the Jewish American Literary Tradition
JWST 1818  (3) Jewish History to 1492
JWST 1828  (3) Jewish History Since 1492
JWST 1830  (3) Global History of Holocaust and Genocide
JWST 1900  (3) Introduction to the Hebrew Bible/Old Testament
JWST 1910  (3) Introduction to the New Testament
JWST 2350  (3) Introduction to Jewish Culture
JWST 2502  (3) Representing the Holocaust
JWST 2551  (3) Modern Jewish Literature
JWST 2600  (3) Judaism, Christianity, and Islam: Abrahamic Religions
JWST 3100  (3) Judaism
JWST 3110  (3) Of Jewish Legends, Folktales and the Supernatural
JWST 3120  (3) Radical Jews
JWST 3150  (3) Jerusalem: The Holy City in History, Legend, and Religious Thought
JWST 3200  (3) Religion and Feminist Thought
JWST 3202  (3) Women, Gender & Sexuality in Jewish Texts & Traditions
JWST 3310  (3) The Bible as Literature
JWST 3401  (3) The Heart of Europe: Filmmakers and Writers in 20th Century Cent

LAND 2004  (3) History of Landscape Architecture
LAND 2100  (6) Studio 1: Foundations of Landscape Architecture
LAND 3003  (3) Site Planning, Materials, and Technologies
LAND 3100  (6) Studio 2: Intermediate Landscape Architecture
LAND 3103  (3) Ecological Planting Design
LAND 4100  (6) Capstone in Landscape Architecture
LAND 4114  (3) Landscape Architecture Theory
LGTC 5010  (2) Second Language Acquisition and Language Education
LGTC 5020  (3) Educational Technology Foundation
LGTC 5030  (2) Language Technology Tools in Practice
LGTC 5031  (2) Emerging Technology for Language Learning
LGTC 5032  (2) Digital Games and Language Learning
LGTC 5035  (2) Online and Blended Language Instruction
LGTC 5040  (2) Telecollaboration for Language Learning Foundations
LGTC 5045  (1) Language Teacher Proficiency Through Telecollaboration
LGTC 5050  (3) Language Technology Practicum
LGTC 5060  (3) Global Voices, Telecollaboration, and Language Learning
LAMS 1000  (3) Introduction to Latin American an

LAWS 7105  (3) Family Law
LAWS 7106  (1-2) Moot Court Competition
LAWS 7111  (3) Contract Theory: Collisions of Contracting and Culture
LAWS 7115  (3) Juvenile Justice
LAWS 7116  (1) Barristers Council
LAWS 7121  (3) Advanced Contracts: Commercial Transactions
LAWS 7122  (2-3) Mining and Mineral Development Law
LAWS 7125  (2) Advanced Domestic Relations
LAWS 7126  (1-2) Transactional Competition
LAWS 7128  (2-4) Jurisprudence
LAWS 7132  (3) Energy, Insecurity, Sustainable Law
LAWS 7135  (3) Parent, Child, and State
LAWS 7138  (3) Legal Philosophy
LAWS 7145  (3) Comparative Family Law
LAWS 7154  (3) Land Use Planning
LAWS 7159  (2) Advanced Trial Advocacy
LAWS 7164  (2) Land Conservation Law
LAWS 7169  (2) Motions Advocacy
LAWS 7200  (3) Anthropology of Law
LAWS 7201  (3) Antitrust
LAWS 7202  (3) Environmental Law
LAWS 7205  (3) Administrative Law
LAWS 7207  (3) Federal Estate and Gift Tax
LAWS 7209  (4) Natural Resources and Environmental Law Clinic
LAWS 7211  (3) Business Planning
LAW

LAWS 8322  (3) Seminar: Environmental Decision Making
LAWS 8325  (2) Seminar: Reforming Criminal Trials
LAWS 8331  (2) Seminar: Equality And Sports
LAWS 8335  (2) Seminar: Advanced Criminal Procedure
LAWS 8341  (3) Seminar: Law and Economics of the Information Age
LAWS 8345  (2) Sem Comp Crim Pro
LAWS 8351  (2) Seminar: Law and Economics of Utility Regulation
LAWS 8355  (2) Seminar: Sentencing Law and Policy
LAWS 8361  (2) Seminar: Advanced Information Privacy
LAWS 8375  (2) Seminar: Advanced Immigration and Citizenship
LAWS 8385  (2) Sem Law & Religion
LAWS 8395  (2) Seminar: Separation of Powers
LAWS 8400  (2) Seminar: Special Topics in International Law
LAWS 8401  (2) Seminar: Securities Litigation and Enforcement
LAWS 8405  (2) Seminar: Public Health Law and Ethics
LAWS 8407  (2) Seminar: Tax Law, Economics and Policy
LAWS 8409  (2) Seminar Special Problems in Conflict Resolution and Management
LAWS 8412  (2) Seminar: Critical Law and Economics
LAWS 8415  (2) Seminar: Bioethics and

MGMT 3100  (3) Operations Management
MGMT 3201  (3) Business Analytics
MGMT 3800  (3) Consulting Skills
MGMT 4110  (3) Supply Chain Management
MGMT 4120  (3) Managing Business Processes
MGMT 4130  (3) Sustainable Operations
MGMT 4140  (3) Project Management
MGMT 4150  (3) International Operations Management
MGMT 4160  (3) Managing Growth: Entrepreneurship and High Growth Ventures
MGMT 4200  (3) Competitive Strategy
MGMT 4240  (3) Competitive Strategy
MGMT 4400  (3) Quality Management
MGMT 4440  (3) Privacy in Big Data Analytics
MGMT 4820  (3) Decision Analytics
MGMT 4825  (3) Experimental Seminar
MGMT 4850  (3) Senior Seminar in Management
MGMT 4900  (1-3) Independent Study
MGMT 4910  (3) OPIM Academic Internship
MGMT 5120  (3) Managing Business Processes
MGMT 5230  (3) Design of Usable Business Systems
MGMT 5820  (3) Decision Analytics
MGMT 8900  (1-3) Independent Study
MGMT 8990  (1-10) Doctoral Dissertation
MKTG 2400  (3) Principles of Marketing for Communications
MKTG 2700  (3) Dig

MATH 6290  (3) Homological Algebra
MATH 6310  (3) Introduction to Real Analysis 1
MATH 6320  (3) Introduction to Real Analysis 2
MATH 6350  (3) Functions of a Complex Variable 1
MATH 6360  (3) Functions of a Complex Variable 2
MATH 6534  (3) Topics in Mathematical Probability
MATH 6550  (3) Introduction to Stochastic Processes
MATH 6730  (3) Set Theory
MATH 6740  (3) Forcing
MATH 6900  (1-3) Independent Study
MATH 6940  (1) Master's Candidate for Degree
MATH 6950  (1-6) Master's Thesis
MATH 8114  (3) Topics in Number Theory
MATH 8174  (3) Topics in Algebra I
MATH 8234  (3) Topics in Differential Geometry
MATH 8250  (3) Mathematical Theory of Relativity 1
MATH 8304  (3) Topics in Analysis 1
MATH 8330  (3) Functional Analysis 1
MATH 8340  (3) Functional Analysis 2
MATH 8370  (3) Harmonic Analysis 1
MATH 8714  (3) Topics in Logic 1 and 2
MATH 8815  (1-3) Ulam Seminar
MATH 8900  (1-3) Independent Study
MATH 8990  (1-10) Doctoral Dissertation
MBAX 6000  (3) Socially Responsible Enterprise
M

MCEN 5147  (3) Mechanobiology
MCEN 5151  (3) Flow Visualization
MCEN 5152  (3) Introduction to Combustion
MCEN 5153  (3) Introduction to Fracture Mechanics
MCEN 5154  (3) Biocolloids and Biomembranes
MCEN 5155  (3) Automated Mechanical Design Synthesis
MCEN 5157  (3) Modeling of Human Movement
MCEN 5161  (3) Aerosols
MCEN 5171  (3) Biofluids on the Micro Scale
MCEN 5173  (3) Finite Element Analysis
MCEN 5174  (3) Failure of Engineering Materials
MCEN 5183  (3) Mechanics of Composite Materials
MCEN 5194  (3) Electrochemical Energy Conversion and Storage
MCEN 5195  (3) Bioinspired Robotics
MCEN 5208  (1-4) Special Topics
MCEN 5228  (1-4) Special Topics in Mechanical Engineering
MCEN 5255  (3) Design for Mfg
MCEN 5258  (1-3) Sp Tpcs-Combustion Seminar
MCEN 5279  (3) Aesthetics in Design
MCEN 5291  (1-2) Project Based Learning in Rural Schools
MCEN 5292  (3) Materials and Devices in Medicine
MCEN 5293  (3) Mechanics of Soft Matter
MCEN 5298  (3) Introduction to Polymers
MCEN 5299  (3) Hous

MCDB 5201  (1) Graduate Lab in Molecular Evolution
MCDB 5210  (3) Cell Structure and Function (Lecture and Discussion)
MCDB 5220  (3) Graduate Core 2
MCDB 5230  (3) Graduate Core 1
MCDB 5250  (3) Topics in Developmental Genetics (Methods and Logic)
MCDB 5301  (3) Immunology
MCDB 5310  (3) Microbial Genetics and Physiology
MCDB 5312  (3-4) Quantitative Optical Imaging
MCDB 5350  (3) Microbial Diversity and the Biosphere
MCDB 5361  (3) Evolution and Development
MCDB 5425  (3) Topics in Membrane Biology: Cell Biology, Physiology and Disease
MCDB 5426  (3) Cell Signaling and Developmental Regulation
MCDB 5427  (3) Biology of the Visual System
MCDB 5441  (4) Animal Developmental Diversity
MCDB 5471  (3) Mechanisms of Gene Regulation in Eukaryotes
MCDB 5520  (3) Bioinformatics and Genomics
MCDB 5521  (1) Bioinformatics and Genomics Laboratory
MCDB 5550  (3) Cells, Molecules and Tissues: A Biophysical Approach
MCDB 5560  (3) Introduction to Biophysics
MCDB 5615  (3) Biology of Stem Cells
MCDB

MUSC 1081  (3) Intensive Music Theory
MUSC 1101  (2) Semester 1 Theory
MUSC 1103  (1) Becoming a Music Teacher
MUSC 1111  (2) Semester 2 Theory
MUSC 1121  (1) Aural Skills Lab, Semester 1
MUSC 1131  (1) Aural Skills Lab, Semester 2
MUSC 1325  (1) Piano Sight Reading
MUSC 1326  (2) Guitar Musicianship and Accompanying
MUSC 1544  (1) Italian Diction
MUSC 1554  (1) English Diction
MUSC 1802  (3) Introduction to Musical Styles and Ideas
MUSC 2041  (1) Applications of Music Technology (5-week segment)
MUSC 2061  (3) Introduction to Music Technology
MUSC 2081  (2) Prepared for the Soundcheck
MUSC 2091  (2) Recording Techniques
MUSC 2101  (2) Semester 3 Theory
MUSC 2103  (2) Introduction to Music Education
MUSC 2111  (2) Semester 4 Theory
MUSC 2121  (1) Aural Skills Lab, Semester 3
MUSC 2131  (1) Aural Skills Lab, Semester 4
MUSC 2325  (2) Applied Harmony for the Keyboard
MUSC 2365  (2) Introduction to Accompanying
MUSC 2608  (1) Alexander Technique
MUSC 2772  (3) World Musics: Asia and Ocean

MUSC 5978  (3) Introduction to Arts Administration
MUSC 5988  (3) The Entrepreneurial Artist
MUSC 6041  (3) Advanced Orchestration
MUSC 6051  (3) Pedagogy of Music Theory
MUSC 6113  (2) Foundations of Music Education
MUSC 6133  (2) Comprehensive Musicianship through Performance
MUSC 6193  (1-3) Selected Studies in Music Education
MUSC 6203  (2) Psychology of Music Learning
MUSC 6213  (2) Assessment of Music Learning
MUSC 6223  (2) Sociology of Music Education
MUSC 6233  (2) Pedagogy of Music Teaching and Learning
MUSC 6243  (1) Applications of Music Pedagogy
MUSC 6325  (2) Seminar in Piano Literature
MUSC 6343  (1) Capstone Portfolio Project
MUSC 6801  (3) Advanced Topics in Music Theory
MUSC 6822  (3) Advanced Studies in Musicology
MUSC 7046  (3) Seminar in Jazz Literature
MUSC 7103  (3) Historical Research in Music Education
MUSC 7113  (3) Quantitative Research in Music Education
MUSC 7138  (3) Contemporary Issues in College Teaching
MUSC 7143  (3) Qualitative Research in Music Educa

PHIL 1000  (3) Introduction to Philosophy
PHIL 1010  (3) Introduction to Western Philosophy: Ancient
PHIL 1020  (3) Introduction to Western Philosophy: Modern
PHIL 1030  (3) Introduction to Global Philosophy
PHIL 1100  (3) Ethics
PHIL 1160  (3) Introduction to Medical Ethics
PHIL 1200  (3) Contemporary Social Problems
PHIL 1250  (3) Poverty, Power, and Patriotism: Issues of Global Justice
PHIL 1350  (3) Knowledge, Mind, and Reality
PHIL 1400  (3) Philosophy and the Sciences
PHIL 1440  (3) Critical Thinking
PHIL 1500  (3) Reading, Writing and Reasoning
PHIL 1600  (3) Philosophy and Religion
PHIL 1700  (3) Philosophy and the Arts
PHIL 1750  (3) Philosophy through Literature
PHIL 1800  (3) Open Topics/Philosophy
PHIL 2140  (3) Environmental Justice
PHIL 2150  (3) Ethics and Sex
PHIL 2160  (3) Ethics and Information Technology
PHIL 2170  (3) Ethics and Economics
PHIL 2200  (3) Major Social and Political Theories
PHIL 2220  (3) Philosophy and Law
PHIL 2240  (3) Philosophy and Sports
PHIL 22

PHYS 7820  (3) Topics in Scientific Writing
PHYS 7830  (1) Plasma Seminar
PHYS 7840  (1-3) Selected Topics for Graduate Independent Study
PHYS 7850  (1-3) Selected Topics for Graduate Independent Study
PHYS 8990  (1-10) Doctoral Dissertation
PLAN 2004  (3) History and Theory of Sustainable Planning and Design
PLAN 2100  (6) Studio 1: Foundations of Planning and Urban Design
PLAN 3005  (3) Process and Practice
PLAN 3100  (6) Studio 2: Intermediate Planning and Urban Design
PLAN 3102  (3) Strategies and Techniques for Sustainable Planning and Design
PLAN 4100  (6) Capstone in Planning and Urban Design
PLAN 4101  (3) Sustainable Futures Planning
PSCI 1101  (3) Introduction to American Politics
PSCI 2004  (3) Survey of Western Political Thought
PSCI 2012  (3) Introduction to Comparative Politics
PSCI 2028  (3) Special Topics
PSCI 2075  (3) Quantitative Research Methods
PSCI 2106  (3) Introduction to Public Policy Analysis
PSCI 2116  (3) Introduction to Environmental Policy and Policy Analy

PORT 1010  (5) Beginning Portuguese 1
PORT 1020  (5) Beginning Portuguese 2
PORT 2110  (3) Second-Year Portuguese 1
PORT 2120  (3) Second-Year Portuguese 2
PORT 2350  (3) Portuguese for Romance Language Speakers
PORT 2800  (3) Brazil: Past and Present
PORT 3003  (3) Advanced Portuguese Language Skills
PORT 3170  (3) Lisbon as a Global City: Cosmopolitanism, Diversity, and Innovation
PORT 3220  (3) Latin American Culture: Spanish America and Brazil
PORT 3230  (3) Transatlantic Relations in the Portuguese Speaking World
PORT 3270  (3) Socio-Environmental Dynamics in Brazil
PORT 4110  (3) Brazilian Literature
PORT 4150  (3) Literature of the Portuguese Speaking World
PORT 4230  (3) Special Topics in Luso-Brazilian and/or African Literature
PORT 4840  (1-3) Independent Study
PORT 5110  (3) Brazilian Literature
PORT 5150  (3) Literature of the Portuguese Speaking World
PRLC 1810  (3) Leadership Foundations and Applications I
PRLC 1820  (3) Leadership Foundations & Applications II
PRLC 2820 

RUSS 1010  (4) Beginning Russian 1
RUSS 1020  (4) Beginning Russian 2
RUSS 1050  (5) Intensive Beginning Russian
RUSS 1900  (1-6) Independent Study
RUSS 2010  (4) Second-Year Russian 1
RUSS 2020  (4) Second-Year Russian 2
RUSS 2121  (3) Topics in Russian and Slavic Culture
RUSS 2211  (3) Russian Culture and Art Under Tsars Great and Terrible
RUSS 2221  (3) Introduction to Modern Russian Culture
RUSS 2222  (3) Sports and the Cold War
RUSS 2231  (3) Fairy Tales of Russia
RUSS 2241  (3) Death and the Undead in Slavic and Nordic Cultures
RUSS 2261  (3) Madness and Gambling: Russian Short Stories of Life on the Edge
RUSS 2271  (3) Space, Invention, and Wonder in Fairy Tales, Literature and Film
RUSS 2311  (3) Energy Cultures: Oil, Coal, and Atoms in Modern Literature and Film
RUSS 2471  (3) Women in Russian Culture: From Folklore to the Nineteenth Century
RUSS 2501  (3) Russia Beyond the Headlines: Media, Politics, Culture, and Environment
RUSS 3000  (3) Advanced Conversation
RUSS 3010  (4)

SOCY 7014  (3) Gender, Race, Class, and Crime
SOCY 7017  (3) Population and Environment
SOCY 7024  (3) Punishment and Social Control
SOCY 7026  (3) Feminist Research Methods
SOCY 7034  (3) Capital Punishment in the United States
SOCY 7036  (3) Feminist Theory
SOCY 7111  (3) Data III--Advanced Data Analysis
SOCY 7121  (3) Qualitative Analysis
SOCY 7131  (3) Seminar in Social Psychology
SOCY 7141  (3) Third-year Paper Seminar
SOCY 7171  (3) Special Topics
SOCY 8991  (1-10) Doctoral Dissertation
SPAN 1000  (3) Cultural Difference through Hispanic Literature
SPAN 1010  (5) Beginning Spanish 1
SPAN 1020  (5) Beginning Spanish 2
SPAN 1150  (5) Intensive First Year Spanish
SPAN 2110  (3) Second-Year Spanish 1
SPAN 2120  (3) Second-Year Spanish 2
SPAN 2150  (5) Intensive Second-Year Spanish
SPAN 2450  (3) Catalan for Spanish Speakers
SPAN 3000  (5) Advanced Spanish Language Skills
SPAN 3001  (3) Spanish Conversation
SPAN 3002  (3) Advanced Spanish Conversation
SPAN 3010  (3) Advanced Rhetoric 

SLHS 7550  (3) Prevention of Hearing Loss from a Public Health Perspective
SLHS 7614  (3) Implantable Devices: Technology and Clinical Application
SLHS 7640  (3) Communication Processes and Hearing Loss: Birth through Six
SLHS 7714  (3) Advanced Topics in Amplification
SLHS 7775  (1) Topics in Cognitive Science
SLHS 7800  (1) Seminars in Audiology: Advanced Topics
SLHS 7849  (1-4) Independent Study 1, PhD
SLHS 7859  (1-4) Independent Study 2, PhD
SLHS 7918  (3) Practicum 3: Clinical Supervision
SLHS 8918  (3) Practicum 3: Classroom Instruction
SLHS 8928  (3) Practicum 3: Research Coordination
SLHS 8990  (1-10) Doctoral Dissertation
STAT 2600  (4) Introduction to Data Science
STAT 3100  (3) Applied Probability
STAT 3400  (3) Applied Regression
STAT 4000  (3) Statistical Methods and Application I
STAT 4010  (3) Statistical Methods and Applications II
STAT 4100  (3) Markov Processes, Queues, and Monte Carlo Simulations
STAT 4230  (3) Stochastic Analysis for Finance
STAT 4250  (3) Data Ass

THTR 6031  (3) On-Stage Studies: American Theatre
THTR 6041  (3) On-Stage Studies: Global Modern Theatre
THTR 6051  (1-3) Production Research and Practicum: Directing
THTR 6071  (3) Seminar: Perspectives on Acting
THTR 6081  (3) Seminar in American Theatre: Lesbians and Gays
THTR 6091  (1-3) Production Research and Practicum: Dramaturgy
THTR 6111  (3) On Stage Studies: Global Contemporary Theatre
THTR 6849  (1-3) Independent Study
THTR 6949  (1) Master's Candidate for Degree
THTR 6959  (1-6) Master's Thesis
THTR 7004  (3-6) Colorado Shakespeare Festival Summer Immersion
THTR 8999  (1-10) Doctoral Dissertation
TDXD 5005  (3) Design Theory
TDXD 5105  (3) Collaboratory in Experience Design 1
TDXD 5500  (3) Experience Design Atelier 1: Design Evolution and Expression
TDXD 5700  (3) Experience Design Atelier 2: Introduction to Design Graphics
TDXD 5805  (3) Professional Portfolio 1
TDXD 6105  (3) Collaboratory in Experience Design 2
TDXD 6210  (3) Storytelling for XD
TDXD 6500  (3) Experien

WRTG 1100  (4) Extended First-Year Writing and Rhetoric
WRTG 1150  (3) First-Year Writing and Rhetoric
WRTG 1160  (3) CMCI First-Year Writing and Rhetoric
WRTG 1250  (3) Advanced First-Year Writing and Rhetoric
WRTG 1840  (1-3) Independent Study in Writing
WRTG 2000  (3) Information and Society
WRTG 2020  (3) Introduction to Creative Nonfiction
WRTG 2090  (3) Electives in Writing
WRTG 2930  (1-6) Internship in Writing and Rhetoric
WRTG 3007  (3) Writing in the Visual Arts
WRTG 3020  (3) Topics in Writing
WRTG 3030  (3) Writing on Science and Society
WRTG 3035  (3) Technical Communication and Design
WRTG 3040  (3) Writing on Business and Society
WRTG 3045  (3) Writing for Emerging Workplaces
WRTG 3090  (1-3) Open Topics in Writing: Advanced
WRTG 3400  (3) Race and Epistemic Justice
WRTG 3840  (1-3) Independent Study
WRTG 3930  (1-6) Internship in Writing and Rhetoric
WRTG 4910  (1-3) Portfolio Curation in Writing and Rhetoric
WRTG 5050  (3) Graduate Studies in Writing and Rhetoric
WRTG 

In [18]:
get_courses_details.head()

Courses  \
1  ACCT 3220  (3) Corporate Financial Reporting 1   
2    ACCT 3225  (6) Corporate Financial Reporting   
3  ACCT 3230  (3) Corporate Financial Reporting 2   
4                  ACCT 3320  (3) Cost Management   
5   ACCT 3440  (3) Income Taxation of Individuals   

                                       full_links  
1  https://catalog.colorado.edu/courses-a-z/acct/  
2  https://catalog.colorado.edu/courses-a-z/acct/  
3  https://catalog.colorado.edu/courses-a-z/acct/  
4  https://catalog.colorado.edu/courses-a-z/acct/  
5  https://catalog.colorado.edu/courses-a-z/acct/

In [19]:
courses_desc['courses_id'] = get_courses_details['Courses'].str.split(' ', n = 1, expand = True)[0].replace(u'\xa0', u' ')
courses_desc['course_name'] = get_courses_details["Courses"].str.split(")", n = 1, expand = True)[1].replace(u'\xa0', u' ')
courses_desc['department_id'] = get_courses_details["Courses"].str.split(" ", n = 1, expand = True)[0].str.split("\xa0", expand = True)[0]

In [20]:
courses_desc.head()

courses_id                       course_name department_id
1  ACCT 3220   Corporate Financial Reporting 1          ACCT
2  ACCT 3225     Corporate Financial Reporting          ACCT
3  ACCT 3230   Corporate Financial Reporting 2          ACCT
4  ACCT 3320                   Cost Management          ACCT
5  ACCT 3440    Income Taxation of Individuals          ACCT

In [21]:
courses_desc.shape

(8862, 3)

In [22]:
courses_desc_dictionary = courses_desc.to_dict('records')

In [23]:
print(courses_desc_dictionary)

[{'courses_id': 'ACCT\xa03220', 'course_name': ' Corporate Financial Reporting 1', 'department_id': 'ACCT'}, {'courses_id': 'ACCT\xa03225', 'course_name': ' Corporate Financial Reporting', 'department_id': 'ACCT'}, {'courses_id': 'ACCT\xa03230', 'course_name': ' Corporate Financial Reporting 2', 'department_id': 'ACCT'}, {'courses_id': 'ACCT\xa03320', 'course_name': ' Cost Management', 'department_id': 'ACCT'}, {'courses_id': 'ACCT\xa03440', 'course_name': ' Income Taxation of Individuals', 'department_id': 'ACCT'}, {'courses_id': 'ACCT\xa03700', 'course_name': ' Accounting in a Global Economy', 'department_id': 'ACCT'}, {'courses_id': 'ACCT\xa04240', 'course_name': ' Advanced Financial Accounting', 'department_id': 'ACCT'}, {'courses_id': 'ACCT\xa04250', 'course_name': ' Financial Statement Analysis', 'department_id': 'ACCT'}, {'courses_id': 'ACCT\xa04540', 'course_name': ' Accounting Information Systems', 'department_id': 'ACCT'}, {'courses_id': 'ACCT\xa04620', 'course_name': ' Audit

## Save the courses_descriptions and Department id to CSV

In [12]:
courses_desc.to_csv("C:/Users/Nikhi/Documents/Big_Data_Architect/Course_Names.csv")
department.to_csv("C:/Users/Nikhi/Documents/Big_Data_Architect/Department_Names.csv")